src/api/todos.js

```diff
@@ -0,0 +1,76 @@
+// Todos table functions 
+import axios from './axios';
+
+// RESTful API endpoint for todo objects
+const endpoint = '/todos';
+
+/**
+ * Get all rows in the todos table.
+ * @returns array of todos 
+ */
+async function getTodos() {
+  const { data } = await axios.get(endpoint);
+  return data;
+}
+
+/**
+ * Insert a new todo in the todos table
+ * @param {*} todo new todo to insert 
+ * @returns true if success, false if not
+ */
+async function postTodo(todo) {
+  const { status } = await axios.post(endpoint, todo);
+  return status === 200;
+}
+
+/**
+ * Update the todo in the todos table. Make sure that id is specified. 
+ * @param {*} todo data to update
+ * @returns true if success, false if not 
+ */
+async function putTodo(todo) {
+  const { status } = axios.put(`${endpoint}/${todo.id}`, todo);
+  return status === 200;
```

Comment:
Again please refer to my previous comment on HTTP semantics for PUT request (editing an existing resource at server) and rest of my notes on the comment